src/App.js

```diff
@@ -1,26 +1,143 @@
-import React from 'react';
-import logo from './logo.svg';
+/*
+ * Main App.
+ */
+
+import React, { Component } from 'react';
+
+// Bulma and custom CSS.
+import 'bulma/css/bulma.css';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+// Data.
+import foods from './foods.json';
+import FoodBoxList from './components/FoodBoxList';
+
+// Components.
+import AddFood from './components/AddFood';
+
+class App extends Component {
+  state = {
+    foodList: foods,
+    searchResult: foods,
+    showForm: false,
+    foodDetails: [],
+    totalCalories: [],
+  };
+
+  handleAdd = (event) => {
+    event.preventDefault();
+    const { foodList, searchResult } = this.state;
+    const { name, calories, image } = event.target;
+
+    const newFood = {
+      name: name.value,
+      calories: calories.value,
+      image: image.value,
+    };
+
+    this.setState({
+      foodList: [...foodList, newFood],
+      searchResult: [...searchResult, newFood],
+      showForm: false,
+    });
+  };
+
+  handleForm = () => {
+    this.setState({
+      showForm: true,
+    });
+  };
+
+  handleSearch = (event) => {
+    const searchText = event.target.value;
+
+    const { foodList } = this.state;
+
+    const searchResult = foodList.filter((food) => {
+      return food.name.toLowerCase().includes(searchText);
+    });
+
+    this.setState({
+      searchResult: searchResult,
+    });
+  };
+
+  handleCalories = (foodInfo, quantity) => {
+    const { totalCalories, foodDetails } = this.state;
+
+    const foodsInfo = {
+      name: foodInfo.name,
+      quantity: Number(quantity),
+      calories: foodInfo.calories * quantity,
+    };
+
+    for (let idx in foodDetails) {
+      if (foodDetails[idx].name === foodsInfo.name) {
+        foodDetails[idx].quantity += Number(quantity);
+
+        Number(foodDetails[idx].quantity);
+
```

Comment:
to follow your logic here (which was a great approach) you wanted now to access the food element that already exists in foodDetails and change it's total calories value. 
Then it is just rendering this foodDetails information in the JSX.